src/lib.rs

```diff
@@ -1,6 +1,455 @@
+use core::panic;
+use std::cmp;
 #[doc = include_str!("../README.md")]
+#[derive(PartialEq, Eq, Debug)]
+pub enum StringDiffOpKind {
+	Substitute(char, char),
+	Insert(char),
+	Delete(char),
+	Transpose(usize, usize),
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct StringDiffOp {
+	pub kind: StringDiffOpKind,
+	pub index: usize,
+}
+
+impl StringDiffOp {
+	// a generic constructor
```

Comment:
Sorry, I meant this comment as an explanation to you in the code review (to help explain the difference between the constructors); we can delete this comment